# Problem 3

Import Word2Vec model, i.e., collection of embedding vectors (vectors.bin file) into spaCy. two short sentences (5 to 6 words each). Compare their similarity using large English model provided by spaCy and imported Word2Vec model. Describe your observations.

In [1]:
import spacy

2023-11-10 21:35:52.114982: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.


In [ ]:
#load space english large model
nlp = spacy.load("en_core_web_lg")

In [7]:
#define sentences to be compared
sen1 = "Every rational potency admits equally of contrary results, but irrational potencies admit of one result only."

In [8]:
sen2 = 'But of beauty, I repeat again that we saw her there shining in company with the celestial forms; and coming to earth we find her here too, shining in clearness through the clearest aperture of sense.'

In [13]:
#create docs for comparison
doc1 = nlp(sen1)
doc2 = nlp(sen2)

In [14]:
#compare sentences with spacy engl model
similarity1 = doc1.similarity(doc2)
print(similarity1)

0.8517517228660552


In [15]:
#load word2vec model that was generated by:  spacy init vectors en vectors.bin /Users/Matt/CS89
nlp2 = spacy.load('/Users/Matt/CS89')

In [16]:
doc1b = nlp2(sen1)
doc2b = nlp2(sen2)

In [17]:
#compare sentences with spacy engl model
similarity2 = doc1b.similarity(doc2b)
print(similarity2)

0.4254609828753514


We see a very drastic difference in the similarity values between the two models.  The spacy large english model rated the sentences from aristotle and plato as nearly 85% similar.  However, the word2vec vectors.bin model only identified the sentences as 42% similar. 